server/src/main/java/org/opensearch/rest/pagination/IndexPaginationStrategy.java

```diff
@@ -0,0 +1,185 @@
+                    : metadata.getIndex().getName().compareTo(requestedToken.lastIndexName) < 0;
+            }
+            return isAscendingSort
+                ? metadata.getCreationDate() > requestedToken.lastIndexCreationTime
+                : metadata.getCreationDate() < requestedToken.lastIndexCreationTime;
+        };
+    }
+
+    private List<IndexMetadata> getMetadataSubList(List<IndexMetadata> sortedIndices, final int pageSize) {
+        if (sortedIndices.isEmpty()) {
+            return new ArrayList<>();
+        }
+        return sortedIndices.subList(0, Math.min(pageSize, sortedIndices.size()));
+    }
+
+    private PageToken getResponseToken(final int pageSize, final int totalIndices, IndexMetadata lastIndex) {
+        if (totalIndices <= pageSize) {
+            return new PageToken(null, DEFAULT_INDICES_PAGINATED_ELEMENT);
+        }
+        return new PageToken(
+            new IndexStrategyToken(lastIndex.getCreationDate(), lastIndex.getIndex().getName()).generateEncryptedToken(),
+            DEFAULT_INDICES_PAGINATED_ELEMENT
+        );
+    }
+
+    @Override
+    @Nullable
+    public PageToken getResponseToken() {
+        return pageToken;
+    }
+
+    @Override
+    public List<String> getRequestedEntities() {
+        return Objects.isNull(requestedIndices) ? new ArrayList<>() : requestedIndices;
+    }
+
+    /**
+     * TokenParser to be used by {@link IndexPaginationStrategy}.
+     * Token would look like: IndexNumberToStartTheNextPageFrom + | + CreationTimeOfLastRespondedIndex + | +
+     * QueryStartTime + | + NameOfLastRespondedIndex
+     */
+    public static class IndexStrategyToken {
+
+        private static final String JOIN_DELIMITER = "|";
+        private static final String SPLIT_REGEX = "\\|";
+        private static final int CREATE_TIME_POS_IN_TOKEN = 0;
+        private static final int INDEX_NAME_POS_IN_TOKEN = 1;
+
+        /**
+         * Represents creation times of last index which was displayed in the page.
+         * Used to identify the new start point in case the indices get created/deleted while queries are executed.
+         */
+        private final long lastIndexCreationTime;
+
+        /**
+         * Represents name of the last index which was displayed in the page.
+         * Used to identify whether the sorted list of indices has changed or not.
+         */
+        private final String lastIndexName;
+
+        public IndexStrategyToken(String requestedTokenString) {
+            validateIndexStrategyToken(requestedTokenString);
+            String decryptedToken = PaginationStrategy.decryptStringToken(requestedTokenString);
+            final String[] decryptedTokenElements = decryptedToken.split(SPLIT_REGEX);
+            this.lastIndexCreationTime = Long.parseLong(decryptedTokenElements[CREATE_TIME_POS_IN_TOKEN]);
+            this.lastIndexName = decryptedTokenElements[INDEX_NAME_POS_IN_TOKEN];
```

Comment:
[minor]: we can combine validate and parsing otherwise we are parsing twice.

Reply:
The validate at RestLayer is for an early return (we may choose to remove it as well). While, using it in the constructor as a safe guard again, incase Rest layer doesn't validate.

If double validating seems misleading, we can choose to remove it from one of the place, i.e.:
1. Skip validating at the Rest layer.
2. Remove from the constructor. 

The only thing is, we didn't want to pass it as an argument to Strategy and keep the behaviour consistent across strategies where they can directly interact with just PageParams, and a data store (ClusterState here).

Please let me know if you feel we should change this.

Reply:
within the constructor itself we can parse at least once

Reply:
got it so we are actually parsing thrice :), for direct transport action call we will need the validation.

Reply:
Added TODO as discussed.